pyomo/solvers/plugins/solvers/GAMS.py

```diff
@@ -1039,6 +1054,49 @@ def solve(self, *args, **kwds):
         return results
 
 
+class OutputStream:
+    """Output stream object for simultaneously writing to multiple streams.
+
+    tee=False:
+        If set writing to this stream will write to stdout.
+    logfile=None:
+        Optionally a logfile can be written.
+
+    """
```

Comment:
I am torn about introducing an entirely new class to manage this, however, I see the merits of having this as a context manager.